org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/WorkspaceManager.java

```diff
@@ -258,7 +258,7 @@ protected void setWorkspaceConfig(IWorkspaceConfig workspaceConfig) {
 	 * Callback after a build was performend
 	 */
 	protected void afterBuild(List<IResourceDescription.Delta> deltas) {
-		for (ILanguageServerAccess.IBuildListener listener : buildListeners) {
+		for (ILanguageServerAccess.IBuildListener listener : new ArrayList<>(buildListeners)) {
```

Comment:
should we use a safe datastructure as ListenerList or CopyOnWriteArraySet as we do it at other places? what do you think @szarnekow ?

Reply:
Yes, I think a CopyOnWriteArrayList (preserving semantics) or CopyOnWriteArraySet would be better to avoid unnecessary copying and also be save wrt adding listeners from read commands.